src/App.tsx

```diff
@@ -1,9 +1,22 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value.toLowerCase().trim());
+  };
+
+  const moviesData = [...moviesFromServer].filter(movie => {
```

Comment:
There's no need to copy an array since the filter method isn't mutating the original array as it creates and returns a copy of an array.